src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1672,15 +1664,9 @@ private boolean validateAuthority(final URL authorityUrl) {
 
             // Set CorrelationId for Instance Discovery
             mDiscovery.setCorrelationId(getRequestCorrelationId());
-            try {
-                boolean result = mDiscovery.isValidAuthority(authorityUrl);
-                Logger.v(TAG, "Finish validating authority:" + authorityUrl + " result:" + result);
-                return result;
-            } catch (Exception exc) {
-                Logger.e(TAG, "Instance validation returned error", "",
-                        ADALError.DEVELOPER_AUTHORITY_CAN_NOT_BE_VALIDED, exc);
-
-            }
+            boolean result = mDiscovery.isValidAuthority(authorityUrl);
```

Comment:
@imakhalova I think that throwing IllegalArgumentException is appropriate from a public developer facing method is the appropriate thing to do when the argument is clearly Illegal.  I think the fact that the passed in authority is null or not a URI should be separated from the case where it is legally formed but not an a trusted authority, which is really what this method is trying to determine.  The fact that is null or not a uri should have been checked at that point that it was passed in originally though.  If that is being re-checked here, we should just remove that redundant check.